data/migrations/20190328174219_recipebook_tables.js

```diff
@@ -0,0 +1,54 @@
+exports.up = function(knex, Promise) {
+  return knex.schema
+    .createTable('dishes', tbl => {
+      tbl.increments();
+      tbl.string('name', 128).notNullable();
+    })
+    .createTable('recipes', tbl => {
+      tbl.increments();
+      tbl
+        .string('name', 128)
+        .notNullable()
+        .unique();
+      tbl.string('instructions', 128).unique();
+      tbl
+        .integer('dish_id')
+        .unsigned()
+        .notNullable()
+        .references('id')
+        .inTable('dishes')
+        .onDelete('CASCADE')
+        .onUpdate('CASCADE');
+    })
+    .createTable('ingredients', tbl => {
+      tbl.increments();
+      tbl.string('name', 128).notNullable();
+    })
+    .createTable('recipe_ingredients', tbl => {
+      tbl.increments();
+      tbl
+        .integer('recipe_id')
+        .unsigned()
+        .notNullable()
+        .references('id')
+        .inTable('recipes')
+        .onDelete('CASCADE')
+        .onUpdate('CASCADE');
+      tbl
+        .integer('ingredients_id')
+        .unsigned()
+        .notNullable()
+        .references('id')
+        .inTable('ingredients')
+        .onDelete('CASCADE')
+        .onUpdate('CASCADE');
+    });
+};
+
+exports.down = function(knex, Promise) {
+  return knex.schema
+    .dropTableIfExists('recipe_ingredients')
+    .dropTableIfExists('ingredients')
+    .dropTableIfExists('recipes')
+    .dropTableIfExists('dishes');
+};
```

Comment:
Great work on the schema - I'm actually starting to think  a single file is actually better now, it kinda feels more like how GQL schema are written and it makes it easier to reference actually in this case.